gateway/router.go

```diff
@@ -134,17 +135,17 @@ func (r *Router) handleReceive() {
 			}
 			msg.Timestamp = time.Now()
 			gw.modifyMessage(&msg)
-			if idx == 0 {
+			if !filesHandled {
 				gw.handleFiles(&msg)
+				filesHandled = true
 			}
 			for _, br := range gw.Bridges {
 				msgIDs = append(msgIDs, gw.handleMessage(&msg, br)...)
 			}
-			// only add the message ID if it doesn't already exists
-			if _, ok := gw.Messages.Get(msg.Protocol + " " + msg.ID); !ok && msg.ID != "" {
-				gw.Messages.Add(msg.Protocol+" "+msg.ID, msgIDs)
-			}
-			idx++
+
+			// Always add/update the message ID. This is necessary as msgIDs
+			// will change if a bridge returns a different ID in response to edits.
+			gw.Messages.Add(msg.Protocol+" "+msg.ID, msgIDs)
 		}
```

Comment:
The `msg.ID  != ""` check is gone though, i'm not sure if this makes a difference.
I'm also not sure if your change is going to break other protocols, so just to be sure could you make this an exception for discord only?
My suggestion is to keep the original lines 143-146 and add a new check below just for discord (or if you're want to test this for the other protocols and it doesn't break anything thats fine too :-)
```
// Always add/update the message ID. This is necessary as msgIDs
// will change if a bridge returns a different ID in response to edits.
if msg.Protocol == "discord"  {
   gw.Messages.Add(msg.Protocol+" "+msg.ID, msgIDs)
}

Reply:
Yes, good catch, the removal of `msg.ID != ""` was an accident.